integrations/aws/main.py

```diff
@@ -0,0 +1,203 @@
+from typing import Any
+
+from fastapi import Response, status
+
+from port_ocean.core.models import Entity
+from aws.utils import (
+    ACCOUNT_ID_PROPERTY,
+    KIND_PROPERTY,
+    REGION_PROPERTY,
+    ResourceKindsWithSpecialHandling,
+    _session_manager,
+    describe_accessible_accounts,
+    batch_resources,
+    describe_single_resource,
+    fix_unserializable_date_properties,
+    get_sessions,
+    get_resource_kinds_from_config,
+    resync_cloudcontrol,
+    update_available_access_credentials,
+    validate_request,
+    get_matching_kinds_from_config,
+)
+from port_ocean.context.ocean import ocean
+from loguru import logger
+from starlette.requests import Request
+from port_ocean.core.ocean_types import ASYNC_GENERATOR_RESYNC_TYPE
+
+
+@ocean.on_resync()
+async def resync_all(kind: str) -> ASYNC_GENERATOR_RESYNC_TYPE:
+    await update_available_access_credentials()
+
+    if kind in iter(ResourceKindsWithSpecialHandling):
+        logger.info("Kind already has a specific handling, skipping", kind=kind)
+        return
+    async for batch in resync_cloudcontrol(kind):
+        yield batch
+
+
+@ocean.on_resync(kind=ResourceKindsWithSpecialHandling.ACCOUNT)
+async def resync_account(kind: str) -> ASYNC_GENERATOR_RESYNC_TYPE:
+    for account in describe_accessible_accounts():
+        yield fix_unserializable_date_properties(account)
+
+
+@ocean.on_resync(kind=ResourceKindsWithSpecialHandling.CLOUDRESOURCE)
+async def resync_generic_cloud_resource(kind: str) -> ASYNC_GENERATOR_RESYNC_TYPE:
+    resource_kinds = get_resource_kinds_from_config(kind)
+
+    if len(resource_kinds) == 0:
+        logger.error(f"Resource kinds not found in port app config for {kind}")
+        return
+
+    for kind in resource_kinds:
+        async for batch in resync_cloudcontrol(kind):
+            yield batch
```

Comment:
no need for fix_unserializable_date_properties?

Reply:
No since `fix_unserializable_date_properties` is used inside `resync_cloudcontrol`

Reply:
Shouldnt we move this outside to keep the code unified?

Reply:
What do you mean by move outside?